### PR TITLE
Fish 28 fixing issue when starting openmq with invalid jre path

### PR DESCRIPTION
**Pull request by breakponchito**
## Description

Fix for the issue when starting the local jms broker using jdk 11
### New tests

I didn't add new unit test because of the complexity to deal with statics when starting the ActiveJmsResourceAdapter class
### Testing Performed

please verify the following comment on jira ticket with the result of manual test that I did on my local environment:
https://payara.atlassian.net/browse/FISH-28?focusedCommentId=57690
### Testing Environment

Windows 10 with JDK 11(zulu), JDK 10(openjdk) and JDK 9(openjdk)

### CODE REVIEW

**Comment by breakponchito** [comment]
Jenkins test

**Comment by breakponchito** [comment]
Jenkins test

File: appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java [hunk 1]
```diff
@@ -1043,9 +1044,13 @@ private void setLifecycleProperties() throws
             if (tmpString == null) {
                 tmpString = "";
             }
-
             String brokerArgs = tmpString;
 
+            //condition to evaluate the JDK and if version greater than 8 then omit the jre folder
+            //by adding the start-args for the broker configuration
+            if (availableJDKForStartArgs()) {
+                brokerArgs = buildStartArgsForJREHome(java_home);
```
Comment (rdebusscher) [inline appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java:1052]: This overrides any defined `start_args` value.  We should be more selective and only add (when not already defined) the `-jrehome` argument

File: appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java [hunk 2]
```diff
@@ -1158,6 +1163,31 @@ private void setLifecycleProperties() throws
...
     }
 
+    /**
+     * This method evaluate the current JDK and return true if the version is greater than 8. This indicator
+     * is used to include the start-args attribute for the broker configuration with the jrehome path attribute
+     *
+     * @return boolean indicator to create the start-args attribute for the broker
+     */
+    private boolean availableJDKForStartArgs() {
+        if (JDK.getMajor() > 8) {
```
Comment (rdebusscher) [inline appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java:1173]: Can be `return JDK.getMajor() > 8;`

File: appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java [hunk 3]
```diff
@@ -1158,6 +1163,31 @@ private void setLifecycleProperties() throws
...
+    }
+
+    /**
+     * This method build the start-args attribute with the value -jrehome for the jms broker configuration
+     *
+     * @param javaHome String with the path to use for the -jrehome attribute
+     * @return String with the formed start-args attribute
+     */
+    private String buildStartArgsForJREHome(String javaHome) {
+        StringBuilder buildStartArgs = new StringBuilder("-jrehome ");
```
Comment (rdebusscher) [inline appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java:1186]: Can be `return "-jrehome "+javaHome;`.

**Review by rdebusscher — CHANGES_REQUESTED** [review]

**Review by Pandrex247 — COMMENTED** [review]
Nothing extra to add on top of what Rudy has suggested

File: appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java (hunk 1 above)
Comment (breakponchito) [reply appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java]: already attended

File: appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java (hunk 2 above)
Comment (breakponchito) [reply appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java]: already attended

File: appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java (hunk 3 above)
Comment (breakponchito) [reply appserver/jms/jms-core/src/main/java/com/sun/enterprise/connectors/jms/system/ActiveJmsResourceAdapter.java]: already attended

**Comment by breakponchito** [comment]
Jenkins test

**Comment by breakponchito** [comment]
Jenkins test

**Review by rdebusscher — APPROVED** [review]
Looks good

**Review by Pandrex247 — APPROVED** [review]

**Pull request merged**